train.py

```diff
@@ -24,6 +26,47 @@ class Config:
     use_action: bool = False
     device: str = 'cuda'
 
+def mask_loss(
+    logit: th.Tensor,
+    mask_gt: th.Tensor,
+    mask_order: Optional[List[Tuple[int, ...]]] = None
+) -> Tuple[th.Tensor, List[Tuple[int, ...]]]:
+    """
+    Get the mask loss for given mask_order and get the best order for the prediction
+    if mask_order is not given, the best permutation for each batch will be used
+    Arguments:
+        logit: [B, K, S1, S2, S3], the mask predictor output logit(K-1 is none)
+        mask_gt: [B, S1, S2, S3], the ground truth mask index(0 is none)
+        mask_order: mask_order[b] is the permutation tuple of object in batch b(length K-1)
+    """
+    def loss_for_permutation(
+        logit: th.Tensor,
+        mask_gt: th.Tensor,
+        permutation: Tuple[int, ...]
+    ) -> th.Tensor:
+        # CrossEntropyLoss on logits is equivalent to NLL on softmax result
+        logit_permuted = th.stack(
+            [logit[b:b+1, -1]] + [logit[b:b+1, i] for i in permutation], dim=1)
+        return nn.CrossEntropyLoss()(logit_permuted, mask_gt[b:b + 1])
```

Comment:
you can probably use `nn.functional` to avoid creating an awkward class instance.

Reply:
Good point, I will consider changing.